adapter/internal/discovery/xds/semanticVersioning.go

```diff
@@ -0,0 +1,403 @@
+			} else {
+				if isLatestMajorVersion {
+					orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier][GetMajorVersionRange(*apiSemVersion)] = *apiSemVersion
+				}
+				orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier][GetMinorVersionRange(*apiSemVersion)] = *apiSemVersion
+			}
+		}
+		// Add the major and/or minor version range matching regexes to the path specifier when
+		// latest major and/or minor version is available
+		for _, route := range orgIDOpenAPIRoutesMap[organizationID][apiIdentifier] {
+			regex := route.GetMatch().GetSafeRegex().GetRegex()
+			regexRewritePattern := route.GetRoute().GetRegexRewrite().GetPattern().GetRegex()
+			apiVersionRegex := GetVersionMatchRegex(apiVersion)
+			if isLatestMajorVersion {
+				regex = strings.Replace(regex, apiVersionRegex, GetMajorMinorVersionRangeRegex(*apiSemVersion), 1)
+				regexRewritePattern = strings.Replace(regexRewritePattern, apiVersionRegex, GetMajorMinorVersionRangeRegex(*apiSemVersion), 1)
+			} else if isLatestMinorVersion {
+				regex = strings.Replace(regex, apiVersionRegex, GetMinorVersionRangeRegex(*apiSemVersion), 1)
+				regexRewritePattern = strings.Replace(regexRewritePattern, apiVersionRegex, GetMinorVersionRangeRegex(*apiSemVersion), 1)
+			}
+			pathSpecifier := &routev3.RouteMatch_SafeRegex{
+				SafeRegex: &envoy_type_matcherv3.RegexMatcher{
+					Regex: regex,
+				},
+			}
+			route.Match.PathSpecifier = pathSpecifier
+			action := &routev3.Route_Route{}
+			action = route.Action.(*routev3.Route_Route)
+			action.Route.RegexRewrite.Pattern.Regex = regexRewritePattern
+			route.Action = action
+		}
+	}
+}
+
+func updateRoutingRulesOnAPIDelete(organizationID, apiIdentifier string, api mgw.MgwSwagger) {
+	// Update the intelligent routing if the deleting API is the latest version of the API range
+	// and the API range has other versions
+	apiRangeIdentifier := GenerateIdentifierForAPIWithoutVersion(api.GetVHost(), api.GetTitle())
+
+	latestAPIVersionMap, latestAPIVersionMapExists := orgIDLatestAPIVersionMap[organizationID][apiRangeIdentifier]
+	if !latestAPIVersionMapExists {
+		return
+	}
+	deletingAPISemVersion, _ := ValidateAndGetVersionComponents(api.GetVersion(), api.GetTitle())
+	if deletingAPISemVersion == nil {
+		return
+	}
+	majorVersionRange := GetMajorVersionRange(*deletingAPISemVersion)
+	newLatestMajorRangeAPIUUID := ""
+	if deletingAPIsMajorRangeLatestAPISemVersion, ok := latestAPIVersionMap[majorVersionRange]; ok {
+		if deletingAPIsMajorRangeLatestAPISemVersion.Version == api.GetVersion() {
+			newLatestMajorRangeAPI := &SemVersion{
+				Version: "",
+				Major:   deletingAPISemVersion.Major,
+				Minor:   0,
+				Patch:   nil,
+			}
+			for apiUUID, swagger := range orgIDAPIMgwSwaggerMap[organizationID] {
+				// Iterate all the API versions other than the deleting API itself
+				if swagger.GetTitle() == api.GetTitle() && apiUUID != apiIdentifier {
```

Comment:
apiIdentifier -> vhost:apiUUID

Reply:
The `apiIdentifier` is already a unique identifier with vhost and apiUUID.

The `range orgIDAPIMgwSwaggerMap[organizationID]` is also providing the same kind of identifier. Hence, renamed `apiUUID` --> `currentApiIdentifier` to avoid confusion